spiketoolkit/preprocessing/remove_artifacts.py

```diff
@@ -22,21 +25,113 @@ def get_traces(self, channel_ids=None, start_frame=None, end_frame=None):
+                        idxs = np.array(range(idx-2, idx+3))
+                    
+                    if np.min(idx) < 0:
+                        idx = idx[idx >= 0]
+                    
+                    median_vals = np.median(traces[:,idxs], axis = 1)
+                    pre_vals = np.vstack((pre_vals,median_vals))
+                    
+                post_vals = np.empty((0,len(traces)),'int32')
+                for idx in iter(post_idx):
+                    if idx == post_idx[0]:
+                        idxs = np.array(range(idx,idx+4))
+                    else:
+                        idxs = np.array(range(idx-2, idx+3))
+                    
+                    if np.max(idx) >= len(traces[0]):
+                        idx = idx[idx < len(traces[0])]
+                    
+                    median_vals = np.median(traces[:,idxs], axis = 1)
+                    post_vals = np.vstack((post_vals,median_vals))
+                
+                interp_traces = np.vstack((pre_vals, post_vals)).T
+                
+                if (self._mode == 'cubic') & (len(all_idx) >= 5):
+                    # Enough fit points present on either side to do cubic spline fit:
+                    interp_function     = sp.interpolate.interp1d(all_idx, interp_traces, self._mode, bounds_error = False, fill_value = 'extrapolate')
+                    traces[:,gap_idx]   = interp_function(gap_idx)
+                elif (self._mode == 'linear') & (len(all_idx) >= 2):
+                    # Enough fit points present for a linear fit
+                    interp_function     = sp.interpolate.interp1d(all_idx, interp_traces, self._mode, bounds_error = False, fill_value = 'extrapolate')
+                    traces[:,gap_idx]   = interp_function(gap_idx)
+                elif len(pre_idx) > len(post_idx):
+                    # not enough fit points, fill with nearest neighbour on side with the most data points
+                    traces[:,gap_idx]   = np.repeat(traces[:,pre_idx[-1]]* np.ones((1,1)),len(gap_idx),0).T
+                elif len(post_idx) > len(pre_idx):
+                    # not enough fit points, fill with nearest neighbour on side with the most data points
+                    traces[:,gap_idx]   = np.repeat(traces[:,post_idx[0]]* np.ones((1,1)),len(gap_idx),0).T
+                elif len(all_idx) > 0:
+                    # not enough fit points, both sides tied for most data points, fill with last pre value
+                    traces[:,gap_idx]   = np.repeat(traces[:,pre_idx[-1]]* np.ones((1,1)),len(gap_idx),0).T
+                else:
+                    # No data to interpolate from on either side of gap;
+                    # Fill with zeros
+                    traces[:,gap_idx] = 0
+                    
         return traces
 
 
-def remove_artifacts(recording, triggers, ms_before=0.5, ms_after=3):
+def remove_artifacts(recording, triggers, ms_before=0.5, ms_after=3, mode = 'zeros', fit_sample_spacing = 1):
```

Comment:
@joramvanrheede Can you add some explanation of the different modes in the docstrings? also `fit_sample_spacing` should be added there :)

Reply:
Yes, will do that now. I also see a 'Travis CI build failed' notification below but I'm unfamiliar with debugging this sort of issue...

Reply:
I updated some stuff on the extractors side and this makes the test failing. I'll fix that. Thanks for the additions!

Give me some other few days to properly test this. In the meanwhile, could you add some mock tests here: https://github.com/SpikeInterface/spiketoolkit/blob/master/spiketoolkit/tests/test_preprocessing.py#L197-L215 ?
Basically similar to what's there already but adding the new option